src/main/java/org/jabref/gui/EntryTypeDialog.java

```diff
@@ -288,10 +291,16 @@ public void actionPerformed(ActionEvent e) {
                 generateButton.setText(Localization.lang("Searching..."));
             });
 
-            Globals.prefs.put(JabRefPreferences.ID_ENTRY_GENERATOR,String.valueOf(comboBox.getSelectedItem()));
-            searchID = idTextField.getText().trim();
-            searchID = searchID.replaceAll(" ", "");
+            Globals.prefs.put(JabRefPreferences.ID_ENTRY_GENERATOR, String.valueOf(comboBox.getSelectedItem()));
+
             fetcher = WebFetchers.getIdBasedFetchers(Globals.prefs.getImportFormatPreferences()).get(comboBox.getSelectedIndex());
+
+            searchID = idTextField.getText();
+            if (!(fetcher instanceof TitleFetcher)) {
```

Comment:
Why do we remove whitespace for the other fetcher? Shouldn't each fetcher decide how to proceed the input? I would just always trim it at this point...

Reply:
All other fetchers have a single alphanumeric id or an url so, trimming in general makes sense.

Reply:
Copy and paste DOIs from PDFs or homepages (e.g., Springer) leads to identifiers. With the fix https://github.com/JabRef/jabref/pull/3697, there is a speedup in the user workflow. One can just copy something from a PDF/homepage and paste it into JabRef and click "Generate".

From https://github.com/koppor/jabref/issues/283:

`http s://doi.org/10.1007/s13222-018-0273-1` and `DOI: 10.1109/MS.2013.97`.

Reply:
I think it really makes sense to add this argument parsing/trimming to the fetcher itself as @tobiasdiez  proposes. Maybe a new method parse input?